### PR TITLE
Fixed firewalld.list_zones for tabbed values

### LINKED ISSUE #60033: [BUG] RHEL 8 salt-call --local firewalld.list_zones traceback
**Description**
RHEL 8 salt-call --local firewalld.list_zones traceback

3002.2
```
[root@localhost ~]# salt-call --local firewalld.list_zones
[ERROR   ] An un-handled exception was caught by salt's global exception handler:
ValueError: not enough values to unpack (expected 2, got 1)
Traceback (most recent call last):
  File "/usr/bin/salt-call", line 11, in <module>
    load_entry_point('salt==3002.2', 'console_scripts', 'salt-call')()
  File "/usr/lib/python3.6/site-packages/salt/scripts.py", line 449, in salt_call
    client.run()
  File "/usr/lib/python3.6/site-packages/salt/cli/call.py", line 58, in run
    caller.run()
  File "/usr/lib/python3.6/site-packages/salt/cli/caller.py", line 112, in run
    ret = self.call()
  File "/usr/lib/python3.6/site-packages/salt/cli/caller.py", line 220, in call
    self.opts, data, func, args, kwargs
  File "/usr/lib/python3.6/site-packages/salt/executors/direct_call.py", line 12, in execute
    return func(*args, **kwargs)
  File "/usr/lib/python3.6/site-packages/salt/modules/firewalld.py", line 125, in list_zones
    (id_, val) = i.strip().split(":")
ValueError: not enough values to unpack (expected 2, got 1)
Traceback (most recent call last):
  File "/usr/bin/salt-call", line 11, in <module>
    load_entry_point('salt==3002.2', 'console_scripts', 'salt-call')()
  File "/usr/lib/python3.6/site-packages/salt/scripts.py", line 449, in salt_call
    client.run()
  File "/usr/lib/python3.6/site-packages/salt/cli/call.py", line 58, in run
    caller.run()
  File "/usr/lib/python3.6/site-packages/salt/cli/caller.py", line 112, in run
    ret = self.call()
  File "/usr/lib/python3.6/site-packages/salt/cli/caller.py", line 220, in call
    self.opts, data, func, args, kwargs
  File "/usr/lib/python3.6/site-packages/salt/executors/direct_call.py", line 12, in execute
    return func(*args, **kwargs)
  File "/usr/lib/python3.6/site-packages/salt/modules/firewalld.py", line 125, in list_zones
    (id_, val) = i.strip().split(":")
ValueError: not enough values to unpack (expected 2, got 1)
```
3003
```
[ec2-user@ip-10-27-83-211 ~]$ sudo salt-call --local firewalld.list_zones
[ERROR   ] An un-handled exception was caught by salt's global exception handler:
ValueError: not enough values to unpack (expected 2, got 1)
Traceback (most recent call last):
  File "/bin/salt-call", line 11, in <module>
    load_entry_point('salt==3003', 'console_scripts', 'salt-call')()
  File "/usr/lib/python3.6/site-packages/salt/scripts.py", line 449, in salt_call
    client.run()
  File "/usr/lib/python3.6/site-packages/salt/cli/call.py", line 58, in run
    caller.run()
  File "/usr/lib/python3.6/site-packages/salt/cli/caller.py", line 112, in run
    ret = self.call()
  File "/usr/lib/python3.6/site-packages/salt/cli/caller.py", line 220, in call
    self.opts, data, func, args, kwargs
  File "/usr/lib/python3.6/site-packages/salt/loader.py", line 1235, in __call__
    return self.loader.run(run_func, *args, **kwargs)
  File "/usr/lib/python3.6/site-packages/salt/loader.py", line 2268, in run
    return self._last_context.run(self._run_as, _func_or_method, *args, **kwargs)
  File "/usr/lib/python3.6/site-packages/contextvars/__init__.py", line 38, in run
    return callable(*args, **kwargs)
  File "/usr/lib/python3.6/site-packages/salt/loader.py", line 2283, in _run_as
    return _func_or_method(*args, **kwargs)
  File "/usr/lib/python3.6/site-packages/salt/executors/direct_call.py", line 12, in execute
    return func(*args, **kwargs)
  File "/usr/lib/python3.6/site-packages/salt/loader.py", line 1235, in __call__
    return self.loader.run(run_func, *args, **kwargs)
  File "/usr/lib/python3.6/site-packages/salt/loader.py", line 2268, in run
    return self._last_context.run(self._run_as, _func_or_method, *args, **kwargs)
  File "/usr/lib/python3.6/site-packages/contextvars/__init__.py", line 38, in run
    return callable(*args, **kwargs)
  File "/usr/lib/python3.6/site-packages/salt/loader.py", line 2283, in _run_as
    return _func_or_method(*args, **kwargs)
  File "/usr/lib/python3.6/site-packages/salt/modules/firewalld.py", line 125, in list_zones
    (id_, val) = i.strip().split(":")
ValueError: not enough values to unpack (expected 2, got 1)
Traceback (most recent call last):
  File "/bin/salt-call", line 11, in <module>
    load_entry_point('salt==3003', 'console_scripts', 'salt-call')()
  File "/usr/lib/python3.6/site-packages/salt/scripts.py", line 449, in salt_call
    client.run()
  File "/usr/lib/python3.6/site-packages/salt/cli/call.py", line 58, in run
    caller.run()
  File "/usr/lib/python3.6/site-packages/salt/cli/caller.py", line 112, in run
    ret = self.call()
  File "/usr/lib/python3.6/site-packages/salt/cli/caller.py", line 220, in call
    self.opts, data, func, args, kwargs
  File "/usr/lib/python3.6/site-packages/salt/loader.py", line 1235, in __call__
    return self.loader.run(run_func, *args, **kwargs)
  File "/usr/lib/python3.6/site-packages/salt/loader.py", line 2268, in run
    return self._last_context.run(self._run_as, _func_or_method, *args, **kwargs)
  File "/usr/lib/python3.6/site-packages/contextvars/__init__.py", line 38, in run
    return callable(*args, **kwargs)
  File "/usr/lib/python3.6/site-packages/salt/loader.py", line 2283, in _run_as
    return _func_or_method(*args, **kwargs)
  File "/usr/lib/python3.6/site-packages/salt/executors/direct_call.py", line 12, in execute
    return func(*args, **kwargs)
  File "/usr/lib/python3.6/site-packages/salt/loader.py", line 1235, in __call__
    return self.loader.run(run_func, *args, **kwargs)
  File "/usr/lib/python3.6/site-packages/salt/loader.py", line 2268, in run
    return self._last_context.run(self._run_as, _func_or_method, *args, **kwargs)
  File "/usr/lib/python3.6/site-packages/contextvars/__init__.py", line 38, in run
    return callable(*args, **kwargs)
  File "/usr/lib/python3.6/site-packages/salt/loader.py", line 2283, in _run_as
    return _func_or_method(*args, **kwargs)
  File "/usr/lib/python3.6/site-packages/salt/modules/firewalld.py", line 125, in list_zones
    (id_, val) = i.strip().split(":")
ValueError: not enough values to unpack (expected 2, got 1)
```
**Setup**
(Please provide relevant configs and/or SLS files (be sure to remove sensitive info).

**Steps to Reproduce the behavior**
(Include debug logs if possible and relevant)

**Expected behavior**

**Screenshots**

**Versions Report**
<details><summary>salt --versions-report</summary>
(Provided by running salt --versions-report. Please also mention any differences in master/minion versions.) 

```
PASTE HERE
```
</details>

**Additional context**

### PR DESCRIPTION
**Pull request by piterpunk**
Some values are returned by `firewall-cmd` indented by `TAB`. These tabbed values breaks `firewalld.list_zones`. The `firewalld.list_all` parses the `firewall-cmd` correctly.

### What does this PR do?
- Splits the `firewall-cmd` zone output parsing to a separated `__parse_zone` function.
- Rewrites `firewalld.list_zones` and `firewalld.list_all` to use the new function.

### What issues does this PR fix or reference?
Fixes: #60033 

### Previous Behavior
```
# salt-call --local firewalld.list_zones
[ERROR   ] An un-handled exception was caught by salt's global exception handler:
ValueError: not enough values to unpack (expected 2, got 1)
Traceback (most recent call last):
  File "/usr/bin/salt-call", line 11, in <module>
    load_entry_point('salt==3003', 'console_scripts', 'salt-call')()
  File "/usr/lib/python3.6/site-packages/salt/scripts.py", line 449, in salt_call
    client.run()
  File "/usr/lib/python3.6/site-packages/salt/cli/call.py", line 58, in run
    caller.run()
  File "/usr/lib/python3.6/site-packages/salt/cli/caller.py", line 112, in run
    ret = self.call()
  File "/usr/lib/python3.6/site-packages/salt/cli/caller.py", line 220, in call
    self.opts, data, func, args, kwargs
  File "/usr/lib/python3.6/site-packages/salt/loader.py", line 1235, in __call__
    return self.loader.run(run_func, *args, **kwargs)
  File "/usr/lib/python3.6/site-packages/salt/loader.py", line 2268, in run
    return self._last_context.run(self._run_as, _func_or_method, *args, **kwargs)
  File "/usr/lib/python3.6/site-packages/contextvars/__init__.py", line 38, in run
    return callable(*args, **kwargs)
  File "/usr/lib/python3.6/site-packages/salt/loader.py", line 2283, in _run_as
    return _func_or_method(*args, **kwargs)
  File "/usr/lib/python3.6/site-packages/salt/executors/direct_call.py", line 12, in execute
    return func(*args, **kwargs)
  File "/usr/lib/python3.6/site-packages/salt/loader.py", line 1235, in __call__
    return self.loader.run(run_func, *args, **kwargs)
  File "/usr/lib/python3.6/site-packages/salt/loader.py", line 2268, in run
    return self._last_context.run(self._run_as, _func_or_method, *args, **kwargs)
  File "/usr/lib/python3.6/site-packages/contextvars/__init__.py", line 38, in run
    return callable(*args, **kwargs)
  File "/usr/lib/python3.6/site-packages/salt/loader.py", line 2283, in _run_as
    return _func_or_method(*args, **kwargs)
  File "/usr/lib/python3.6/site-packages/salt/modules/firewalld.py", line 125, in list_zones
    (id_, val) = i.strip().split(":")
ValueError: not enough values to unpack (expected 2, got 1)
Traceback (most recent call last):
  File "/usr/bin/salt-call", line 11, in <module>
    load_entry_point('salt==3003', 'console_scripts', 'salt-call')()
  File "/usr/lib/python3.6/site-packages/salt/scripts.py", line 449, in salt_call
    client.run()
  File "/usr/lib/python3.6/site-packages/salt/cli/call.py", line 58, in run
    caller.run()
  File "/usr/lib/python3.6/site-packages/salt/cli/caller.py", line 112, in run
    ret = self.call()
  File "/usr/lib/python3.6/site-packages/salt/cli/caller.py", line 220, in call
    self.opts, data, func, args, kwargs
  File "/usr/lib/python3.6/site-packages/salt/loader.py", line 1235, in __call__
    return self.loader.run(run_func, *args, **kwargs)
  File "/usr/lib/python3.6/site-packages/salt/loader.py", line 2268, in run
    return self._last_context.run(self._run_as, _func_or_method, *args, **kwargs)
  File "/usr/lib/python3.6/site-packages/contextvars/__init__.py", line 38, in run
    return callable(*args, **kwargs)
  File "/usr/lib/python3.6/site-packages/salt/loader.py", line 2283, in _run_as
    return _func_or_method(*args, **kwargs)
  File "/usr/lib/python3.6/site-packages/salt/executors/direct_call.py", line 12, in execute
    return func(*args, **kwargs)
  File "/usr/lib/python3.6/site-packages/salt/loader.py", line 1235, in __call__
    return self.loader.run(run_func, *args, **kwargs)
  File "/usr/lib/python3.6/site-packages/salt/loader.py", line 2268, in run
    return self._last_context.run(self._run_as, _func_or_method, *args, **kwargs)
  File "/usr/lib/python3.6/site-packages/contextvars/__init__.py", line 38, in run
    return callable(*args, **kwargs)
  File "/usr/lib/python3.6/site-packages/salt/loader.py", line 2283, in _run_as
    return _func_or_method(*args, **kwargs)
  File "/usr/lib/python3.6/site-packages/salt/modules/firewalld.py", line 125, in list_zones
    (id_, val) = i.strip().split(":")
ValueError: not enough values to unpack (expected 2, got 1)
```

### New Behavior
```
# salt-call --local firewalld.list_zones
local:
    ----------
    block:
        ----------
        forward-ports:
        icmp-block-inversion:
            - no
        icmp-blocks:
        interfaces:
        masquerade:
            - no
        ports:
        protocols:
        rich rules:
        services:
        source-ports:
        sources:
        target:
            - %%REJECT%%
    dmz:
        ----------
        forward-ports:
        icmp-block-inversion:
            - no
        icmp-blocks:
        interfaces:
        masquerade:
            - no
        ports:
        protocols:
        rich rules:
        services:
            - ssh
        source-ports:
        sources:
        target:
            - default
    drop:
        ----------
        forward-ports:
        icmp-block-inversion:
            - no
        icmp-blocks:
        interfaces:
        masquerade:
            - no
        ports:
        protocols:
        rich rules:
        services:
        source-ports:
        sources:
        target:
            - DROP
    external:
        ----------
        forward-ports:
        icmp-block-inversion:
            - no
        icmp-blocks:
        interfaces:
        masquerade:
            - yes
        ports:
        protocols:
        rich rules:
        services:
            - ssh
        source-ports:
        sources:
        target:
            - default
    home:
        ----------
        forward-ports:
        icmp-block-inversion:
            - no
        icmp-blocks:
        interfaces:
        masquerade:
            - no
        ports:
        protocols:
        rich rules:
        services:
            - cockpit dhcpv6-client mdns samba-client ssh
        source-ports:
        sources:
        target:
            - default
    internal:
        ----------
        forward-ports:
        icmp-block-inversion:
            - no
        icmp-blocks:
        interfaces:
        masquerade:
            - no
        ports:
        protocols:
        rich rules:
        services:
            - cockpit dhcpv6-client mdns samba-client ssh
        source-ports:
        sources:
        target:
            - default
    nm-shared:
        ----------
        forward-ports:
        icmp-block-inversion:
            - no
        icmp-blocks:
        interfaces:
        masquerade:
            - no
        ports:
        protocols:
            - icmp ipv6-icmp
        rich rules:
            - rule priority="32767" reject
        services:
            - dhcp dns ssh
        source-ports:
        sources:
        target:
            - ACCEPT
    public:
        ----------
        forward-ports:
        icmp-block-inversion:
            - no
        icmp-blocks:
        interfaces:
        masquerade:
            - no
        ports:
        protocols:
        rich rules:
        services:
            - cockpit dhcpv6-client ssh
        source-ports:
        sources:
        target:
            - default
    trusted:
        ----------
        forward-ports:
        icmp-block-inversion:
            - no
        icmp-blocks:
        interfaces:
        masquerade:
            - no
        ports:
        protocols:
        rich rules:
        services:
        source-ports:
        sources:
        target:
            - ACCEPT
    work:
        ----------
        forward-ports:
            - port=22:proto=tcp:toport=2222:toaddr=10.0.0.10
        icmp-block-inversion:
            - no
        icmp-blocks:
        interfaces:
        masquerade:
            - no
        ports:
        protocols:
        rich rules:
        services:
            - cockpit dhcpv6-client ssh
        source-ports:
        sources:
        target:
            - default
```

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**
- [x] Changelog - https://docs.saltstack.com/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
No

### CODE REVIEW

**Comment by piterpunk** [comment]
re-run pr-centos-7-x86_64-py3-m2crypto-pytest

**Comment by piterpunk** [comment]
@sagetherage it's possible to see this one merged for `Silicon`?

**Comment by sagetherage** [comment]
@piterpunk I have updated the label and it needs a review and to pass tests or get a few more reviews -- I will see what I can do to help :)

File: tests/unit/modules/test_firewalld.py [hunk 1]
```diff
@@ -40,8 +35,73 @@ def test_list_zones(self):
         """
         Test for List everything added for or enabled in all zones
         """
-        with patch.object(firewalld, "__firewall_cmd", return_value=[]):
```
Comment (Ch3LL) [inline tests/unit/modules/test_firewalld.py:43]: I still think there is value in checking what happens if firewall command does not return anything. Could we include this and the test below you edited to be in a seperate test where we are testing an empty response.

**Review by Ch3LL — CHANGES_REQUESTED** [review]

File: tests/unit/modules/test_firewalld.py (hunk 1 above)
Comment (piterpunk) [reply tests/unit/modules/test_firewalld.py:43]: I added a test for empty response in `list_zones` and restored the one for `list_all`. 

But this specific test was originally wrong, probably by some copy'n'paste error, it said that will test `list_zones` but calls `default_zone`. With this mistake, the original tests didn't tests `list_zones`. The code path for an empty `default_zone` is already tested in the `test_default_zone`, which only returns the `__firewall_cmd` output as-is. I don't think it's needed.

**Comment by Ch3LL** [comment]
Just looks to be a merge conflict that needs to be cleaned up and then we can get this one merged in

**Comment by piterpunk** [comment]
> Just looks to be a merge conflict that needs to be cleaned up and then we can get this one merged in

Ok, I will fix this tonight.

Thank you!

**Comment by piterpunk** [comment]
re-run pr-windows-2019-x64-py3-pytest

**Comment by piterpunk** [comment]
@Ch3LL , I hope now it's good to merge.

**Review by Ch3LL — APPROVED** [review]
